lms/djangoapps/support/views/course_entitlements.py

```diff
@@ -0,0 +1,133 @@
+"""
+Support tool for changing and granting course entitlements
+"""
+from django.contrib.auth.models import User
+from django.db.models import Q
+from django.http import HttpResponseBadRequest
+from django.utils.decorators import method_decorator
+from edx_rest_framework_extensions.authentication import JwtAuthentication
+from rest_framework import permissions, status, viewsets
+from rest_framework.response import Response
+from six import text_type
+
+from entitlements.api.v1.permissions import IsAdminOrAuthenticatedReadOnly
+from entitlements.api.v1.serializers import SupportCourseEntitlementSerializer
+from entitlements.models import CourseEntitlement, CourseEntitlementSupportDetail
+from openedx.core.djangoapps.cors_csrf.authentication import SessionAuthenticationCrossDomainCsrf
+from lms.djangoapps.support.decorators import require_support_permission
+
+
+class EntitlementSupportView(viewsets.ModelViewSet):
+    """
+    Allows viewing and changing learner course entitlements, used the support team.
+    """
+    authentication_classes = (JwtAuthentication, SessionAuthenticationCrossDomainCsrf,)
+    permission_classes = (permissions.IsAuthenticated, IsAdminOrAuthenticatedReadOnly,)
+    queryset = CourseEntitlement.objects.all()
+    serializer_class = SupportCourseEntitlementSerializer
+
+    def filter_queryset(self, queryset):
+        try:
+            username_or_email = self.request.GET.get('user')
+            user = User.objects.get(Q(username=username_or_email) | Q(email=username_or_email))
+            queryset = queryset.filter(user=user)
+            return super(EntitlementSupportView, self).filter_queryset(queryset).order_by('created')
+        except (KeyError, User.DoesNotExist):
+            return queryset.order_by('created')
+
+    @method_decorator(require_support_permission)
+    def update(self, request):
+        """ Allows support staff to unexpire a user's entitlement."""
+        support_user = request.user
+        try:
+            username_or_email = self.request.data['user']
+            user = User.objects.get(Q(username=username_or_email) | Q(email=username_or_email))
+        except User.DoesNotExist:
+            return HttpResponseBadRequest(
+                u'Could not find user {user}'.format(user=username_or_email)
+            )
+        try:
+            course_uuid = request.data['course_uuid']
```

Comment:
I think we should talk about using course_uuid and stuff here. If the support user is updating an entitlement, the support page will know the entitlement uuid, so using that to grab the entitlement would probably be more direct.

Reply:
It would also hopefully allow us to avoid a lot of these possible exceptions.